### ISSUE TITLE
'bytearray' object has no attribute 'encode' 

### ISSUE THREAD

**Issue by WaltzKing** [zsquareplusc/python-msp430-tools#2]
Trying to run msp430.bsl5.uart and getting the error "'bytearray' object has no attribute 'encode'".  This is with Python 2.7.12.  Is this a Python2 vs. Python3 issue?

C:\work>python -m msp430.bsl5.uart -p com11 -v testhex3.txt
Debug is False
Verbosity level set to 2
Python version: 2.7.12 (v2.7.12:d33e0cf91556, Jun 27 2016, 15:19:22) [MSC v.1500 32 bit (Intel)]
pySerial version: 3.4
action list:
   program_file()
   reset()
Programming...
Write segment at 0x4600 64 bytes

An error occurred:
'bytearray' object has no attribute 'encode'
Cleaning up after error...

**Comment by psheph** [comment]
I do not believe this is a Python2 vs Python3 issue. I see the same failure in Python 3.6. The error message has changed slightly to:
"AttributeError: 'bytes' object has no attribute 'encode'"
